### ISSUE TITLE
Read release notes from a changelog file

### ISSUE THREAD

**Issue by laughedelic** [ohnosequences/sbt-github-release#26]
See https://keepachangelog.com/en/1.0.0/. 

Many projects prefer to keep a changelog in one file rather than separate release notes in `notes/<version>.markdown` supported by this plugin. I would be very useful to add some simple parsing of the changelog file to extract all release notes. Could be used either to publish single release on Github, or to sync all releases (see #19).

----

There is another similar initialive: [Conventional Changelog](https://github.com/conventional-changelog/conventional-changelog), which is more common in JS projects. It is accompanied with a toolchain to write "conventional commits", generate the changelog from them and publish releases. This is a good motivating example showing how other communities solve the same problem.

**Comment by ldeluigi** [comment]
In addition to that, overriding `ghreleaseNotes` doesn't work. And this project seems dead.

**Comment by laughedelic** [comment]
@ldeluigi `ghreleaseNotes` worked for me last time I used it. Notice that it's a function, not just a string.

But you are right that this project is not maintained. I don't have access to the repository anymore.

**Comment by ldeluigi** [comment]
In my build.sbt I assign it with 
`ghreleaseNotes := (_ => IO.read(baseDirectory.value / "release_notes.md")),`

But it uses the default value anyway.



> @ldeluigi `ghreleaseNotes` worked for me last time I used it. Notice that it's a function, not just a string.
> 
> But you are right that this project is not maintained. I don't have access to the repository anymore.

**Comment by laughedelic** [comment]
Maybe it's a scoping issue. Do you override it in a subproject, or in `ThisBuild` or somewhere else? You can run `inspect ghreleaseNotes` to see where does the definition come from and how the value is delegated. Also maybe `inspect tree ...` and `inspect actual ...`.

**Comment by ldeluigi** [comment]
Ty! Yeah I indeed found out it was a scoping problem. Then the plugin broke... I think it could be outdated:

```
Error:  org.kohsuke.github.HttpException: {"message":"Resource not accessible by integration","documentation_url":"https://docs.github.com/rest/reference/users#get-the-authenticated-user"}
Error:  	at org.kohsuke.github.Requester.handleApiError(Requester.java:695)
Error:  	at org.kohsuke.github.Requester._to(Requester.java:298)
Error:  	at org.kohsuke.github.Requester.to(Requester.java:239)
Error:  	at org.kohsuke.github.GitHub.getMyself(GitHub.java:383)
Error:  	at org.kohsuke.github.GitHub.<init>(GitHub.java:157)
Error:  	at org.kohsuke.github.GitHubBuilder.build(GitHubBuilder.java:207)
Error:  	at org.kohsuke.github.GitHub.connectUsingOAuth(GitHub.java:218)
Error:  	at ohnosequences.sbt.GithubRelease$defs$.$anonfun$ghreleaseGetRepo$1(GithubRelease.scala:74)
Error:  	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
Error:  	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
Error:  	at sbt.std.Transform$$anon$4.work(Transform.scala:67)
Error:  	at sbt.Execute.$anonfun$submit$2(Execute.scala:281)
Error:  	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
Error:  	at sbt.Execute.work(Execute.scala:290)
Error:  	at sbt.Execute.$anonfun$submit$1(Execute.scala:281)
Error:  	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
Error:  	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
Error:  	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
Error:  	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
Error:  	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
Error:  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
Error:  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
Error:  	at java.lang.Thread.run(Thread.java:748)
Error:  Caused by: org.kohsuke.github.HttpException: Server returned HTTP response code: 403, message: 'Forbidden' for URL: https://api.github.com/user
Error:  	at org.kohsuke.github.Requester.parse(Requester.java:638)
[...]
```

**Comment by laughedelic** [comment]
This looks like missing/wrong credentials. Check https://github.com/ohnosequences/sbt-github-release#credentials

**Comment by ldeluigi** [comment]
> This looks like missing/wrong credentials. Check https://github.com/ohnosequences/sbt-github-release#credentials

I was using Github Actions that defines the GITHUB_TOKEN automatically

**Comment by laughedelic** [comment]
In Github Actions you have the token secret, but you need to expose it through `env`. For example:

```yaml
      - name: Publish Github release
        env:
          GITHUB_TOKEN: ${{ github.token }}
        run: sbt githubRelease
```

**Comment by laughedelic** [comment]
Also, if you're using Github Actions, I recommend you to use https://github.com/actions/create-release action instead of this plugin. If you generate your changelog in `baseDirectory.value / "release_notes.md"` as you wrote above, you can have this step:

```yaml
      - name: Create Release
        uses: actions/create-release@v1
        env:
          GITHUB_TOKEN: ${{ github.token }}
        with:
          tag_name: ${{ github.ref }} # if you tagged, or just pass an explicit version
          body_path: 'release_notes.md'
```

You can also use one of the many changelog-generating actions and feed its output to this action.

**Comment by ldeluigi** [comment]
Thanks, I thought the same, but I ended up using https://github.com/marvinpinto/action-automatic-releases, which automates some inputs and the changelog by reading the commit history. Maybe I'll switch to that, but for now I'm already set.

Given that I'm using [dynver](https://github.com/dwijnand/sbt-dynver) for calculating the software version, I needed to export it as environment variable from the corresponding dynver sbt task. This took some time to figure it out but I made it at the end.

Ty for all the help.

Now the only thing left, which I can't really do for now, at least I think so, is releasing the software to multiple different repositories, in my case Github Packages + Sonatype (for now it's only on sonatype). But I gave up as it's a useless thing to do. But I would have liked it anyway.

In other words I'm looking for an action that releases on Github packages.
